detail.py

```diff
@@ -0,0 +1,11 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from catalog import models as cmod
+
+@view_function
+def process_request(request, p:cmod.Product = None):
+
+    context = {
```

Comment:
here you would do something to get the product and attach it to the request for the LastFiveMiddleware. Something like this:
if product in request.last_five:
        request.last_five.remove(product)
request.last_five.insert(0, product)

Reply:
Thank you for the suggestion. I think this kind of change would be appropriate.